scripts/hyperkit.mjs

```diff
@@ -9,26 +9,83 @@ import path from 'path';
 // The version of hyperkit to build
 const ver = 'v0.20210107';
 
-async function run() {
+/**
+ * Build the Hyperkit binary.
+ * @param workPath {string} The directory to work in; must be empty.
+ * @returns {string} The executable in the work directory.
+ */
+async function buildHyperkit(workPath) {
   // Using git and make to build the binary is intentional. There is no binary
   // download available from the project. Minikube checks the hyperkit version
   // so the correct version information needs to be included in the binary. Make
   // is used by the project to build a binary and it assumes the project was
   // retrieved via git and that git metadata for the version is available. The
   // Makefile uses git to retrieve the version and the sha (which is used for an
   // internal assertion).
-  const destFile = path.resolve(process.cwd(), 'resources', os.platform(), 'hyperkit');
-  const sourceDir = await fs.mkdtemp(path.join(os.tmpdir(), 'hyperkit-'));
+  await spawn('git', 'clone', '--depth', '1', '--branch', ver, 'https://github.com/moby/hyperkit.git', workPath);
+  await spawn('make', '-C', workPath);
+  const outPath = path.resolve(workPath, 'build', 'hyperkit');
+
+  await fs.promises.chmod(outPath, 0o755);
+
+  return outPath;
+}
+
+/**
+ * Build the docker-machine driver binary.
+ * @param workPath {string} The directory to work in; must be empty.
+ * @returns {string} The executable in the work directory.
+ */
+async function buildDockerMachineDriver(workPath) {
+  const project = 'docker-machine-driver-hyperkit';
+  const version = 'v2.0.0-alpha.2';
+  const url = `https://github.com/rancher-sandbox/${ project }/releases/download/${ version }/${ project }`;
+  const outPath = path.join(workPath, project);
+
+  await spawn('curl', '-Lo', outPath, url);
+  console.log('Sudo privileges require to set docker-machine driver suid');
+  await spawn('sudo', 'chown', 'root:wheel', outPath);
```

Comment:
I'm not sure the user always sees the `sudo` prompt.

It appeared to me that `npm i` was hanging. I stopped it with ctrl-C, ran `sudo echo x`, reran `npm i` and it quickly finished.

Reply:
Hmm, right, the display here can be some distance, because things are asynchronous:
```
Sudo privileges require to set docker-machine driver suid
Password:Downloading https://get.helm.sh/helm-v3.5.2-darwin-amd64.tar.gz to /var/folders/fp/rb1j0g8j65d8pdlyw_7zxt8w0000gn/T/rd-helm-Macde9/helm.tar.gz...
x darwin-amd64/
x darwin-amd64/helm
x darwin-amd64/LICENSE
x darwin-amd64/README.md
Downloading https://github.com/rancher/kim/releases/download/v0.1.0-alpha.10/kim-darwin-amd64 to /Users/markyen/src/work/src/github.com/rancher-sandbox/rd/resources/darwin/bin/kim...

(invisible prompt here)
```

Reply:
Can you do a synchronous `sudo echo x` check at the start to check if the user needs to grant the process sudo privileges?

Reply:
Changed in bc8448cbefa55161d7ac947225e49cf9f529177a to make `sudo` itself display the prompt, so that it isn't lost.  Also changing it to a single `sudo` invocation, in case the user has disabled password caching.